pkg/operator/rules.go

```diff
@@ -42,31 +43,50 @@ const (
 
 type PrometheusRuleSelector struct {
 	ruleFormat   RuleConfigurationFormat
+	version      semver.Version
 	ruleSelector labels.Selector
 	nsLabeler    *namespacelabeler.Labeler
 	ruleInformer *informers.ForResource
 	logger       log.Logger
 }
 
-func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, labelSelector *metav1.LabelSelector, nsLabeler *namespacelabeler.Labeler, ruleInformer *informers.ForResource, logger log.Logger) (*PrometheusRuleSelector, error) {
+func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, version string, labelSelector *metav1.LabelSelector, nsLabeler *namespacelabeler.Labeler, ruleInformer *informers.ForResource, logger log.Logger) (*PrometheusRuleSelector, error) {
+	componentVersion, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse Prometheus version")
```

Comment:
(nit) The caller should already log information about the component.
```suggestion
		return nil, errors.Wrap(err, "failed to parse version")
```